src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FeaturedTableService.cs

```diff
@@ -0,0 +1,194 @@
+using static GovUk.Education.ExploreEducationStatistics.Admin.Validators.ValidationErrorMessages;
+using Unit = GovUk.Education.ExploreEducationStatistics.Common.Model.Unit;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services;
+
+public class FeaturedTableService : IFeaturedTableService
+{
+    private readonly ContentDbContext _contentDbContext;
+    private readonly IPersistenceHelper<ContentDbContext> _persistenceHelper;
+    private readonly IDataBlockService _dataBlockService;
+    private readonly IUserService _userService;
+    private readonly IMapper _mapper;
+
+    public FeaturedTableService(ContentDbContext contentDbContext,
+        IPersistenceHelper<ContentDbContext> persistenceHelper,
+        IDataBlockService dataBlockService,
+        IUserService userService,
+        IMapper mapper)
+    {
+        _contentDbContext = contentDbContext;
+        _persistenceHelper = persistenceHelper;
+        _dataBlockService = dataBlockService;
+        _userService = userService;
+        _mapper = mapper;
+    }
+
+    public async Task<Either<ActionResult, FeaturedTableViewModel>> Get(Guid releaseId, Guid dataBlockId)
+    {
+        return await _persistenceHelper.CheckEntityExists<Release>(releaseId)
+            .OnSuccess(_userService.CheckCanViewRelease)
+            .OnSuccess(async _ =>
+                await _persistenceHelper.CheckEntityExists<FeaturedTable>(query =>
+                    query.Where(ft => ft.DataBlockId == dataBlockId && ft.ReleaseId == releaseId)))
+            .OnSuccess(featuredTable => _mapper.Map<FeaturedTableViewModel>(featuredTable));
+    }
+
+    public async Task<Either<ActionResult, List<FeaturedTableViewModel>>> List(Guid releaseId)
+    {
+        return await _persistenceHelper.CheckEntityExists<Release>(releaseId)
+            .OnSuccess(_userService.CheckCanViewRelease)
+            .OnSuccess(async release =>
+            {
+                var featuredTableList = await ListFeaturedTables(release.Id);
+                return _mapper.Map<List<FeaturedTableViewModel>>(featuredTableList);
+            });
+    }
+
+    public async Task<Either<ActionResult, FeaturedTableViewModel>> Create(Guid releaseId,
+        FeaturedTableCreateRequest request)
+    {
+        return await _persistenceHelper.CheckEntityExists<Release>(releaseId)
+            .OnSuccess(_userService.CheckCanUpdateRelease)
+            .OnSuccess(async _ =>
+                await _persistenceHelper.CheckEntityExists<DataBlock>(request.DataBlockId))
+            .OnSuccessDo(async dataBlock =>
+            {
+                var hasFeaturedTable = await _contentDbContext.FeaturedTables
+                    .AnyAsync(ft => ft.DataBlockId == dataBlock.Id);
+                if (hasFeaturedTable)
+                {
+                    return new Either<ActionResult, Unit>(
+                        ValidationUtils.ValidationActionResult(DataBlockAlreadyHasFeaturedTable));
+                }
```

Comment:
I think FeaturedTables and DataBlocks now have a one to one relationship - but I've not tested it at all - so might be best to leave this for me to test on Thursday

Reply:
I'm guessing that after you set the one to one relationship it's the `unique: true` change being added which you saw?

![image](https://github.com/dfe-analytical-services/explore-education-statistics/assets/4147126/3fe4ede9-ea9a-4f42-a503-fcc0e88a65f5)

When I originally left this comment I was thinking of a unique index for the `DataBlockId` and `ReleaseId` combination so that additional featured tables can be created for the same data block as long as they are on different releases, but given that we clone data blocks when amending releases and they get new id's then this is even better.

Reply:
I'm guessing that after you set the one to one relationship it's the `unique: true` change being added which you saw?

![image](https://github.com/dfe-analytical-services/explore-education-statistics/assets/4147126/3fe4ede9-ea9a-4f42-a503-fcc0e88a65f5)

When I originally left this comment I was thinking of a unique index for the `DataBlockId` and `ReleaseId` combination so that additional featured tables can be created for the same data block as long as they are on different releases, but given that we clone data blocks when amending releases and they get new id's then this is even better.

Reply:
> I'm guessing that after you set the one to one relationship it's the unique: true change being added which you saw?

That's right.